integrations/react.md

```diff
@@ -3,107 +3,151 @@ layout: default
+    import { Editor } from '@tinymce/tinymce-react';
 
-## 3. Setup `react-tinymce`
+    class App extends React.Component {
+      handleEditorChange = (e) => {
+        console.log('Content was updated:', e.target.getContent());
+      }
 
-Install the npm package and save it to your `package.json` with `--save`.
+      render() {
+        return (
+          <Editor
+            initialValue="<p>This is the initial content of the editor</p>"
+            init={% raw %}{{{% endraw %}
+              height: 500,
+              menubar: false,
+              plugins: [
+                'advlist autolink lists link image charmap print preview anchor',
+                'searchreplace visualblocks code fullscreen',
+                'insertdatetime media table paste code help wordcount'
+              ],
+              toolbar:
+                'undo redo | formatselect | bold italic backcolor | \
+                alignleft aligncenter alignright alignjustify | \
+                bullist numlist outdent indent | removeformat | help'
+            {% raw %}}}{% endraw %}
+            onChange={this.handleEditorChange}
+          />
+        );
+      }
+    }
 
-```
-$ npm install --save @tinymce/tinymce-react
-```
+    export default App;
+    ```
+    This JavaScript file will create the class `App` containing a TinyMCE editor configured to replicate the example on the [Basic example page]({{site.baseurl}}/docs/demo/basic-example/).
+6. Provide access to TinyMCE using the Tiny Cloud, TinyMCE Self-hosted or by bundling TinyMCE using a module loader.
 
-### 3.1 Loading TinyMCE
-#### Auto-loading from TinyMCE Cloud
-`@tinymce/tinymce-react` requires `tinymce` to be globally available to work, but to make it as easy as possible it will automatically load [TinyMCE Cloud](https://www.tinymce.com/docs/cloud-deployment-guide/editor-and-features/) if it can't find TinyMCE available when the component is mounting. To get rid of the `This domain is not registered...` warning, sign up for the cloud and enter the api key like this:
+    * **Tiny Cloud**
 
-```js
-<Editor apiKey='YOUR_API_KEY' init={% raw %}{{{% endraw %} /* your other settings */ {% raw %}}}{% endraw %} />
-```
+        Include the `apiKey` option in the editor element and include your [TinyMCE API key]({{site.baseurl}}/signup/).
 
-You can also define what cloud channel you want to use out these three:
+        Such as:
 
-* `stable` **Default**. The most stable and well-tested version that has passed the Tiny quality assurance process.
-* `testing` This channel will deploy the current candidate for release to the `stable` channel.
-* `dev` The cutting edge version of TinyMCE updated daily for the daring users.
+        ```js
+        <Editor apiKey='your-api-key' init={% raw %}{{{% endraw %} /* your other settings */ {% raw %}}}{% endraw %} />
+        ```
+        _(Optional)_ The `cloudChannel` option can be used to connect to one of the following cloud channels:
```

Comment:
This section feels very specific and detailed, and seems a little odd within this document?

Reply:
it is. attempting to add the technical information at this time, I'll move this information in the process.